### DISCUSSION TITLE
How To Update A Node?

### DISCUSSION THREAD

**Discussion by kkowal89** [pnp/powershell#177] [Q&A] [answered]
I am using PowerShell 7.0 and trying to change the audienceIds for a navigationNode.

```
$node = Get-PnPNavigationNode -Id 2105
$node.AudienceIds

```
My problem is that there is no update-function for my $node-object.

```
$node.AudienceIds = [guid]
$node.update() -> does not exist
```

Is there another way to update this field with the audiences with PowerShell 7.0?

When I compare the PowerShell 5 and PowerShell 7 i realize that one node object is from type:
Deserialized.Microsoft.SharePoint.Client.NavigationNode and the other 
Microsoft.SharePoint.Client.NavigationNode

I try to use it with PowerShell 7 cause I use it with Azure Automation Function. It only allows PowerShell 7.

**Comment by erwinvanhunen** [comment, marked as answer]
I just checked but have no problems with calling the update method on the node object in PowerShell 7.0.3, PowerShell 7.1.1 nor PowerShell 5. In all environments the node object is being returned a NavigationNode object.

Are you running the code in a separate job within your function? And does that job return the node? PowerShell can serialize and deserialize the object between a job and the main calling script, which could explain the Deserialized. prefix in PS 7 in your Azure function.

**Reply by kkowal89** [reply]
Hello Erwin!

Thanks for your fast response. I found the solution just in the moment you wrote me.

I was using the -UseWindowsPowerShell param while loading the module. This was legacy, cause I had code 400 errors when I tried to Connect to my tenant.

Now I registered the Azure AD application on my tenant and loaded the module without the -UseWindowsPowerShell param. 

Now when I load the node its giving me back a Object and not the deserialized version of it. And most important I can use the update() function on my node object. So I can now add audienceIds to my Navigation!

Thank you! You are doing good work here!

**Reply by erwinvanhunen** [reply]
Please mark this item as answered, it help us keep track of the status of questions.
